### PR TITLE
Update SAMLSSOServiceProviderdAOImpl to use named parameters

### PR DESCRIPTION
**Pull request by Osara-B**
### Proposed changes in this pull request

This PR introduces usage of named parameters in SAMLSSOServiceProviderDAOImpl.java. To support the transition from positional parameters to named parameters SAMLSSOServiceProvider.java is updated. This change improves code readability and maintainability.

-The refactored code replaces positional indices with named parameters.
-SQL queries are being executed using named parameters instead of positional indices.

Related to,
* https://github.com/wso2/product-is/issues/20992

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAOImpl.java [hunk 1]
```diff
@@ -329,8 +329,8 @@ private boolean processIsServiceProviderExists(Connection connection, String iss
 
         try (NamedPreparedStatement statement = new NamedPreparedStatement(connection,
                 SAMLSSOServiceProviderConstants.SqlQueries.GET_SAML2_SSO_CONFIG_ID_BY_ISSUER)) {
-            statement.setString(1, issuer);
-            statement.setString(2, tenantUUID);
+            statement.setString("ISSUER", issuer);
```
Comment (darshanasbg) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAOImpl.java:332]: Lets use constants.

Comment (Osara-B) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAOImpl.java]: Resolved in [Update SAMLSSOServiceProviderDAOImpl and SAMLSSOServiceProviderConsta…](https://github.com/wso2/carbon-identity-framework/pull/5887/commits/827ebcb2fe58af8cbfe730557521122d0bafa0f1)

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderConstants.java [hunk 2]
```diff
@@ -24,67 +24,138 @@ private SAMLSSOServiceProviderConstants() {
 
     }
 
+    public static class SAML2TableColumns {
```
Comment (ImalshaG) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderConstants.java:27]: Let's have an empty line here.

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderConstants.java [hunk 3]
```diff
@@ -24,67 +24,138 @@ private SAMLSSOServiceProviderConstants() {
...
+        private SAML2TableColumns() {
+
+        }
+
+        //IDN_SAML2_SSO table
```
Comment (darshanasbg) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderConstants.java:32]: Add a space after `//`

Comment (Osara-B) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderConstants.java]: Resolved in [Update SAMLSSOServiceProviderDAOImpl and SAMLSSOServiceProviderConsta…](https://github.com/wso2/carbon-identity-framework/pull/5887/commits/f02e487e0cdf790f28d84ff94eab068d4d397e32)

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderConstants.java (hunk 2 above)
Comment (Osara-B) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderConstants.java:27]: Resolved in [Update SAMLSSOServiceProviderDAOImpl and SAMLSSOServiceProviderConsta…](https://github.com/wso2/carbon-identity-framework/pull/5887/commits/f02e487e0cdf790f28d84ff94eab068d4d397e32)

**Review by darshanasbg — APPROVED** [review]

**Pull request merged**